tracer/src/Datadog.Trace/Propagators/ContextPropagationHeaderStyle.cs

```diff
@@ -0,0 +1,66 @@
+﻿// <copyright file="ContextPropagationHeaderStyle.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+namespace Datadog.Trace.Propagators;
+
+/// <summary>
+/// Values used to indicate propagation header styles.
+/// </summary>
+internal static class ContextPropagationHeaderStyle
```

Comment:
Is there a particular reason for uppercasing the string names? Uppercase or lowercase shouldn't really matter, but it looks like OTEL specifies the names as lowercase as environment variable values so that might be more commonplace: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration

Reply:
I think I was going to use `ToUpperInvariant()` on the input before comparing them, but now I'm not sure if I went that way. I'll take another look.

Reply:
This made me realize I need to test that the settings are actually not case-sensitive :)

Reply:
I've changed the logic in `SpanContextPropagatorFactory` to use `string.Equals()` with `StringComparison.OrdinalIgnoreCase` instead of a `switch` statement and fixed the const strings in this file.

I also added `SpanContextPropagatorFactoryTests` to test the case-sensitivity.